app/controllers/dish_ingredients_controller.rb

```diff
@@ -0,0 +1,10 @@
+class DishIngredientsController < ApplicationController
+  def create
+    @dish = Dish.find(params[:dish_id])
+    @ingredient = Ingredient.where("name ILIKE ?", "%#{params[:ingredient]}%").first
+  
+    DishIngredient.create!(dish_id: @dish.id, ingredient_id: @ingredient.id)
+  
+    redirect_to "/dishes/#{@dish.id}"
+    end
+end
```

Comment:
Aww, man... no `where` in the controller. We only want to do CRUD AR in the controller. For this all you needed to do was `find` the ingredient by the provided ingredient ID.